kinto_client/importer.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+import argparse
+import getpass
+import logging
+import os
+from six import iteritems
+
+from kinto_client import Client
+from kinto_client.exceptions import KintoException
+
+COMMAND_LOG_LEVEL = 25
+logging.addLevelName(COMMAND_LOG_LEVEL, 'COMMAND')
+logging.basicConfig(level=COMMAND_LOG_LEVEL, format="%(message)s")
+global_logger = logging.getLogger(__file__)
+
+
+def is_same_record(fields, one, two):
+    for key in fields:
+        if one.get(key) != two.get(key):
+            return False
+    return True
+
+
+class KintoImporter(object):
+    # Default configure parser values
+    all_default_parameters = False
+    remote_server = None
+    authentication = None
+    files = None
+    verbosity = None
+
+    # Remote Permissions
+    bucket_permissions = None
+    collection_permissions = None
+
+    # Default values
+    default_host = "http://localhost:8888/v1"
+    default_bucket = None
+    default_collection = None
+    default_auth = None
+    default_files = None
+
+    # Sync flags
+    create = True
+    update = True
+    delete = True
+
+    def __init__(self, logger=None, arguments=None,
+                 local_client=None, remote_client=None,
+                 *args, **kwargs):
+        if not hasattr(self, 'record_fields'):
+            raise ValueError(
+                "%r: record_fields attribute is not defined." % self)
+
+        self.logger = logger or global_logger
+
+        self._local_records = None
+        self._remote_records = None
+
+        parser = self.configure_parser(*args, **kwargs)
+        self.args = self.get_arguments(parser, args=arguments)
+        self.setup_logger()
+        self.local_client = self.setup_local_client(local_client)
+        self.remote_client = self.setup_remote_client(remote_client)
+
+    def configure_parser(self, parser=None,
+                         all_default_parameters=None, remote_server=None,
+                         authentication=None, files=None, verbosity=None,
+                         *args, **kwargs):
+        """Return an argparse.ArgumentParser pre-configured object."""
```

Comment:
The docstring should contain some more information about the arguments it expects, and what it does with the extra args/kwargfs.